upgrade/v12xto133/upgrade.go

```diff
@@ -0,0 +1,56 @@
+package v12xto133
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
+)
+
+const (
+	upgradeLogPrefix = "upgrade from v1.2.x to v1.3.3: "
+)
+
+func UpgradeResources(namespace string, lhClient *lhclientset.Clientset, kubeClient *clientset.Clientset) error {
+	if err := upgradeVolumes(namespace, lhClient); err != nil {
+		return err
+	}
+	return nil
+}
+
+func upgradeVolumes(namespace string, lhClient *lhclientset.Clientset) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, upgradeLogPrefix+"upgrade volume failed")
+	}()
+
+	volumes, err := lhClient.LonghornV1beta2().Volumes(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, "failed to list all existing Longhorn volumes during the volume upgrade")
+	}
+
+	for _, v := range volumes.Items {
+		if v.Spec.RestoreVolumeRecurringJob != "" {
+			continue
+		}
+
+		v.Spec.RestoreVolumeRecurringJob = longhorn.RestoreVolumeRecurringJobDefault
```

Comment:
Is it possible to put the conversion in the mutator?

Reply:
Ok never mind, the mutator is only effective during create and update.

Reply:
Got it